qiskit_experiments/calibration/experiments/spectroscopy.py

```diff
@@ -0,0 +1,317 @@
+
+    __analysis_class__ = SpectroscopyAnalysis
+
+    # Supported units for spectroscopy.
+    __units__ = {"Hz": 1.0, "kHz": 1.0e3, "MHz": 1.0e6, "GHz": 1.0e9}
+
+    # default run options
+    __run_defaults__ = {"meas_level": MeasLevel.CLASSIFIED}
+
+    def __init__(
+        self, qubit: int, frequency_shifts: Union[List[float], np.array], unit: Optional[str] = "Hz"
+    ):
+        """
+        A spectroscopy experiment run by shifting the frequency of the qubit.
+        The parameters of the GaussianSquare spectroscopy pulse are specified at run-time.
+        The spectroscopy pulse has the following parameters:
+        - amp: The amplitude of the pulse must be between 0 and 1, the default is 0.1.
+        - duration: The duration of the spectroscopy pulse in samples, the default is 1000 samples.
+        - sigma: The standard deviation of the pulse, the default is 5 x duration.
+        - width: The width of the flat-top in the pulse, the default is 0, i.e. a Gaussian.
+
+        Args:
+            qubit: The qubit on which to run spectroscopy.
+            frequency_shifts: The frequencies to scan in the experiment.
+            unit: The unit in which the user specifies the frequencies. Can be one
+                of 'Hz', 'kHz', 'MHz', 'GHz'. Internally, all frequencies will be converted
+                to 'Hz'.
+
+        Raises:
+            ValueError: if there are less than three frequency shifts or if the unit is not known.
+
+        """
+        if len(frequency_shifts) < 3:
+            raise ValueError("Spectroscopy requires at least three frequencies.")
+
+        if unit not in self.__units__:
+            raise ValueError(f"Unsupported unit: {unit}.")
+
+        self._frequency_shifts = [freq * self.__units__[unit] for freq in frequency_shifts]
+
+        super().__init__([qubit], circuit_options=("amp", "duration", "sigma", "width"))
+
+    # pylint: disable=unused-argument
+    def circuits(self, backend: Optional["Backend"] = None, **circuit_options):
+        """
+        Create the circuit for the spectroscopy experiment. The circuits are based on a
+        GaussianSquare pulse and a frequency_shift instruction encapsulated in a gate.
+
+        Args:
+            backend: A backend object.
+            circuit_options: Key word arguments to run the circuits. The circuit options are
+                - amp: The amplitude of the GaussianSquare pulse, defaults to 0.1.
+                - duration: The duration of the GaussianSquare pulse, defaults to 10240.
+                - sigma: The standard deviation of the GaussianSquare pulse, defaults to one
+                    fith of the duration.
+                - width: The width of the flat top in the GaussianSquare pulse, defaults to 0.
+
+        Returns:
+            circuits: The circuits that will run the spectroscopy experiment.
+        """
+
+        amp = circuit_options.get("amp", 0.1)
+        duration = circuit_options.get("duration", 1024)
```

Comment:
Can you specify SI unit also for duration and sigma? Perhaps `dt` is uncommon unit for experimentalists.

Reply:
I think it should be fairly common but I added some extra info. See here: 2937795